src/libs/ReportUtils.js

```diff
@@ -1237,6 +1237,27 @@ function getDisplayNamesStringFromTooltips(displayNamesWithTooltips) {
     ).join(', ');
 }
 
+/**
+ * @param {String} reportID
+ * @param {Object} [actionsToMerge]
+ * @return {Object}
+ */
+function getLastVisibleMessage(reportID, actionsToMerge = {}) {
+    const report = getReport(reportID);
+    const lastVisibleAction = ReportActionsUtils.getLastVisibleAction(reportID, actionsToMerge);
+
+    // For Chat Report, let us return [Deleted message] when the last visible action
+    // is a deleted parent which has visible child actions
+    if (ReportActionsUtils.isDeletedParentAction(lastVisibleAction) && isChatReport(report)) {
+        return {
+            lastMessageText: Localize.translateLocal('parentReportAction.deletedMessage'),
+        };
+    }
+
+    // Fetch the last visible message for report represented by reportID and based on actions to merge.
+    return ReportActionsUtils.getLastVisibleMessage(reportID, actionsToMerge);
+}
```

Comment:
What's the purpose of creating separate function in ReportUtils?
Can this logic be added in `ReportActionsUtils.getLastVisibleMessage` directly?

Reply:
> What's the purpose of creating separate function in ReportUtils?
> Can this logic be added in `ReportActionsUtils.getLastVisibleMessage` directly?

@situchan We did try this in [this commit here](https://github.com/Expensify/App/pull/28364/commits/ae59e9bb0640b33cf07632d1047dc0f18509b5d3). But, it resulted in lint issues which is the reason why we landed up here.